docs/CONTRIBUTING.md

```diff
@@ -1,45 +1,57 @@
-
-2. You express interest in the issue on GitHub.
-
-    We try to ensure that multiple people are not working on the same issue -- at least not without being aware of it.
+2. Express interest in the issue on GitHub.
+    - We try to ensure that multiple people are not working on the same issue.
  
 3. We discuss the issue and how best to fix it.
-
-    The idea is to come up with an implementation plan and to hash out any design choices.
+    - The idea is to come up with an implementation plan and to hash out any design choices.
 
 4. You fork the Flix repository to your own account on GitHub.
 
-5. You clone this repository to your local machine.
+5. You clone the forked repository to your own machine.
 
 6. You make the required changes, add test cases, documentation, and so forth.
 
 7. You push the local changes to your GitHub repository and create a new pull request (PR).
-   - See [Pull Requests](PULLREQUEST.md) for information on formatting
+   - See below for instructions on how to make a good pull request.
 
 8. GitHub automatically compiles and tests the PR. Be sure to check the results!
 
 9. We perform a code review and iterate on the implementation.
+   - You should expect that most PRs will require multiple iterations.
+
+**Note:** You must agree to release your contributions under the Apache 2.0 license. You affirm this by adding yourself to AUTHORS.md.
 
-10. If the code is of sufficient quality it is merged in!
+## Pull Requests
 
-*Note:* 
+- Pull requests should use [Semantic Commit Messages](https://gist.github.com/joshbuchea/6f47e86d2510bce28f8e7f42ae84c716).
+    - We use the tags: `feat`, `fix`, `docs`, `style`, `refactor`, `test`, `chore`.
+    - The summary should be a single clear sentence (all lower case except for names).
+- Pull requests should be a small as possible.
+    - An ideal pull request is between one and a few hundred lines of code.
+    - A pull request should focus on a single change.
+    - Most features are broken down into multiple pull requests.
+- When you create a pull request, click `Files changed` to ensure that everything looks as you expect.
+    - The easier a pull request is to understand the faster it will be merged. A win-win.
+- If the pull request fixes a bug the summary should end with (fixes #9999).
+- If the pull request is related to a ticket the summary should end with (related to #9999).
 
-You must agree to release your contributions under the Apache 2.0 license.
+## Code Quality
 
-You affirm this by adding yourself to AUTHORS.md.
+We are interested in correct, well-documented, and well-tested code. 
 
+The argument that "the code works" is not sufficient. It must be maintainable. It must follow our style.
 
+We refer to [STYLE.md](./STYLE.md) for more information on code style.
+- As an example of high-quality code, we refer to `FastBoolUnification.scala`.
 
+Getting your code merged into Flix is a privilege, not a right.
```

Comment:
Can we tone this down slightly? Maybe just "We expect all PRs to follow the established style and to be maintainable..."

The way it is now feels overly combative.